server/middleware/auth.js

```diff
@@ -1,29 +1,34 @@
-const crypto = require('crypto');
-const jwt = require('jsonwebtoken');
 const User = require('../models/user');
 const { ForbiddenError } = require('../models/error');
 const getLogger = require('../libs/logger');
+const passport = require('passport');
+const { KeycloakClient } = require('../libs/auth');
+const { cacheUserToken } = require('../middleware/userTokenCache');
 
+const loginMiddleware = passport.authenticate('keycloak');
 
-function authMiddleware(req, res, next) {
-    const logger = getLogger(req.requestId);
-    if (req.get('X-Auth-Token')) {
-        if (!req.user) {
-            req.user = new User({
-                username: req.get('X-Auth-Username'),
-                id: req.get('X-Auth-UserId'),
-                groups: req.get('X-Auth-Groups'),
-                roles: req.get('X-Auth-Roles'),
-                email: req.get('X-Auth-Email'),
-                uuid: req.get('X-Auth-Subject')
-            });
+const loginCallbackMiddleware = passport.authenticate('keycloak', {
+    //TODO: is there a better way to handle this redirect?
+    successRedirect: 'back',
```

Comment:
It should be possible to pass the redirect URL to Keycloak and have it pass it back in the call back